packages/object/src/index.ts

```diff
@@ -87,6 +109,21 @@ export class TopologyObject implements ITopologyObject {
 	// biome-ignore lint: value can't be unknown because of protobuf
 	callFn(fn: string, args: any) {
 		const vertex = this.hashGraph.addToFrontier({ type: fn, value: args });
+		// the vertex certainly has dependencies
+		let past: Set<Hash> =
+			this.states.get(vertex.dependencies[0])?.past || new Set();
```

Comment:
```suggestion
		let past: Set<Hash> = new Set();
```
it can help make the logic less complex